dvc/repo/data.py

```diff
@@ -0,0 +1,259 @@
+import os
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, TypedDict, cast
+
+if TYPE_CHECKING:
+    from scmrepo.base import Base
+
+    from dvc.output import Output
+    from dvc.repo import Repo
+    from dvc_data.hashfile.db import HashFileDB
+    from dvc_data.hashfile.obj import HashFile
+
+
+def _in_cache(obj: Optional["HashFile"], cache: "HashFileDB") -> bool:
+    from dvc_objects.errors import ObjectFormatError
+
+    if not obj:
+        return False
+    if not obj.hash_info.value:
+        return False
+
+    try:
+        cache.check(obj.hash_info.value)
+        return True
+    except (FileNotFoundError, ObjectFormatError):
+        return False
+
+
+def _shallow_diff(
+    root: str,
+    old_obj: Optional["HashFile"],
+    new_obj: Optional["HashFile"],
+    cache: "HashFileDB",
+) -> Dict[str, List[str]]:
+    # TODO: add support for shallow diff in dvc-data
+    # TODO: we may want to recursively do in_cache check
+    d = {}
+
+    from dvc_data.objects.tree import Tree
+
+    if isinstance(new_obj, Tree):
+        root = os.path.sep.join([root, ""])
+
+    if not _in_cache(old_obj, cache):
+        d["not_in_cache"] = [root]
+
+    if old_obj is None and new_obj is None:
+        return d
+    if old_obj is None:
+        return {"added": [root], **d}
```

Comment:
I thought of one more thing: This should be called `new` I think, because you can `dvc add` an updated file and it will be listed as `modified`.

Reply:
hmm, it's interesting Git does "new file" ... VS Code does "added"

Reply:
I am not sure about this one tbh.

Reply:
> VS Code does "added"

I see `A` (added) in both Explorer and Source Control views:

<img width="417" alt="image" src="https://user-images.githubusercontent.com/1477535/191179382-fd127c23-58a5-481c-8053-d5d2bddd6470.png">

<img width="390" alt="image" src="https://user-images.githubusercontent.com/1477535/191179504-0ad82aa7-ce35-43a6-86ec-5ede0600ca6f.png">